internal/taskstore/store.go

```diff
@@ -55,6 +55,14 @@ func WithTimeProvider(tp TimeProvider) Option {
 	}
 }
 
+func (s *Mem) SetAuthenticator(a Authenticator) {
+	s.authenticator = a
+}
+
+func (s *Mem) SetTimeProvider(p TimeProvider) {
+	s.timeProvider = p
+}
```

Comment:
Seems like this is only needed to enable testutil, I'd rather reorganize test code, we shouldn't be encouraging changing dependencies on a created object in non-test code.